### PR TITLE
Changed GET methods for loans

### LINKED ISSUE #67: Use blockchain controller to return loan phase

### PR DESCRIPTION
**Pull request by ricardomilhazes**
- Rebuilt all GET methods for loans
- Changed full_name to public variable
- Changed URLs as well

Please, check this one with attention! It's working but there might me better ways to do some functionalities.
It's been a long night so something might have slipped

### CODE REVIEW

**Review by ricardofsc10 — APPROVED** [review]
Gj!

File: tuichain/api/views/loans.py [hunk 1]
```diff
@@ -226,7 +227,7 @@ def validate_loan_request(request, id):
     ).get()
 
     loan.identifier = str(result.identifier)
-    loan.state = 1
+    loan.state = 2
```
Comment (albertofaria) [inline tuichain/api/views/loans.py]: Would be preferable to use something like `LoanState.APPROVED.value`.

File: tuichain/api/views/loans.py [hunk 2]
```diff
@@ -320,18 +321,28 @@ def get_loan_request(request, id):
             status=HTTP_404_NOT_FOUND,
         )
 
+    loan_dict = loan.to_dict()
+
+    if loan.state == 2:
```
Comment (albertofaria) [inline tuichain/api/views/loans.py]: Would be preferable to use something like `LoanState(loan.state) == LoanState.APPROVED`. Ditto for other occurrences.

File: tuichain/api/views/loans.py [hunk 3]
```diff
@@ -6,13 +6,14 @@
...
-from tuichain.api.models import Loan, Investment
+from tuichain.api.models import Loan, Profile
 from tuichain.api.enums import LoanState
 from tuichain.api.services.blockchain import controller
-from tuichain_ethereum import Address
+from tuichain_ethereum import Address, LoanIdentifier, LoanPhase
 from rest_framework.permissions import *
 from rest_framework.decorators import api_view, permission_classes
 import decimal
+import numpy as np
```
Comment (albertofaria) [inline tuichain/api/views/loans.py]: No need to depend on numpy.

File: tuichain/api/views/loans.py [hunk 4]
```diff
@@ -348,8 +359,20 @@ def get_personal_loan_requests(request):
     user = request.user
 
     loan_list = Loan.objects.filter(student=user)
-
-    result = [obj.to_dict() for obj in loan_list]
+    result = np.empty(shape=loan_list.count(), dtype=object)
```
Comment (albertofaria) [inline tuichain/api/views/loans.py]: Why add a dependency on numpy just to create a list of placeholder objects? One possibility would be to start with an empty list and using `.append()` to append to it as needed. Ditto for other occurrences.

File: tuichain/api/views/loans.py [hunk 5]
```diff
@@ -363,80 +386,42 @@ def get_personal_loan_requests(request):
 
 @api_view(["GET"])
 @permission_classes((IsAuthenticated,))
-def get_all_loan_requests(request):
+def get_operating_loans(request):
```
Comment (albertofaria) [inline tuichain/api/views/loans.py:429]: What does "operating" mean here? Is there an existing term that we can use instead?

**Review by albertofaria — CHANGES_REQUESTED** [review]

File: tuichain/api/views/loans.py (hunk 5 above)
Comment (ricardomilhazes) [reply tuichain/api/views/loans.py:429]: Operating means that they aren't rejected or withdrawn
Means that they still are up and running basically

Comment (albertofaria) [reply tuichain/api/views/loans.py:429]: What about canceled or expired loans?

**Comment by albertofaria** [comment]
Should drop the changes to `requirements.txt`.

File: tuichain/api/views/loans.py (hunk 5 above)
Comment (ricardomilhazes) [reply tuichain/api/views/loans.py:429]: Frontend told me they didn't need canceled and expired loans, but I can add that method as well, no problem

Comment (albertofaria) [reply tuichain/api/views/loans.py:429]: The trouble is that this endpoint will return them as well.

File: tuichain/api/views/loans.py [hunk 6]
```diff
@@ -340,6 +340,9 @@ def get_loan(request, id):
         )
         loan_dict["funded_value_atto_dai"] = funded_value_atto_dai
 
+        token_contact_adress = controller.loans.get_by_identifier(LoanIdentifier(loan.identifier)).token_contract_adress()
+        loan_dict["token_contract_address"] = str(token_contact_adress)
```
Comment (albertofaria) [inline tuichain/api/views/loans.py]: I'd rename `token_contract_address` to `token_address` in `loan_dict`.

File: tuichain/api/views/loans.py [hunk 7]
```diff
@@ -340,6 +340,9 @@ def get_loan(request, id):
         )
         loan_dict["funded_value_atto_dai"] = funded_value_atto_dai
 
+        token_contact_adress = controller.loans.get_by_identifier(LoanIdentifier(loan.identifier)).token_contract_adress()
```
Comment (albertofaria) [inline tuichain/api/views/loans.py]: Should factor out all three occurrences of `controller.loans.get_by_identifier(LoanIdentifier(loan.identifier))` into a variable.

Comment (ricardomilhazes) [reply tuichain/api/views/loans.py]: Yep, makes sense

**Review by albertofaria — APPROVED** [review]

**Pull request merged**